Index: linux-4.4.14/drivers/net/phy/micrel.c
===================================================================
--- linux-4.4.14.orig/drivers/net/phy/micrel.c
+++ linux-4.4.14/drivers/net/phy/micrel.c
@@ -261,6 +261,7 @@ static int kszphy_config_init(struct phy
 	struct kszphy_priv *priv = phydev->priv;
 	const struct kszphy_type *type;
 	int ret;
+	int temp = 0;
 
 	if (!priv)
 		return 0;
@@ -281,8 +282,13 @@ static int kszphy_config_init(struct phy
 		}
 	}
 
-	if (priv->led_mode >= 0)
+	if (priv->led_mode >= 0) {
+		/* disable PME */
+		temp = phy_read(phydev, 0x16);
+		temp &= ~(1 << 15);
+		phy_write(phydev, 0x16, temp);
 		kszphy_setup_led(phydev, type->led_mode_reg, priv->led_mode);
+	}
 
 	return 0;
 }
@@ -599,7 +605,10 @@ static int kszphy_probe(struct phy_devic
 	} else {
 		priv->led_mode = -1;
 	}
-
+	
+	/* bypass device tree */
+	priv->led_mode = 0;
+	
 	clk = devm_clk_get(&phydev->dev, "rmii-ref");
 	/* NOTE: clk may be NULL if building without CONFIG_HAVE_CLK */
 	if (!IS_ERR_OR_NULL(clk)) {
